### PR TITLE
Unix-timestamps for lamport clocks

### LINKED ISSUE #57: Document Lamport unix time bidding process
Currently  https://github.com/status-im/specs/blob/978c5b77bb45805df91a326132b4bc89671006d2/status-payloads-spec.md#clock-vs-timestamp-and-message-ordering mentions use of Lamport clocks.

However, it doesn't appear to mention the unix time bidding process tweak and rationale (with Lamport invariant) described in https://github.com/status-im/status-react/blob/c9994b5d0f72377ec51000541e6b02500f8430f5/src/status_im/utils/clocks.cljs

;; But what we can do, is to use our time to make a "bid", hoping that it will
;; beat the current chat-timestamp. So our Lamport timestamp format is:
;; {unix-timestamp-ms}{2-digits-post-id}

cc @corpetty @cammellos

### PR DESCRIPTION
**Pull request by 0kok0**
Closes  #57.

### CODE REVIEW

File: status-payloads-spec.md [hunk 1]
```diff
@@ -173,16 +173,27 @@ The following messages types MUST be supported:
 
 #### Clock vs Timestamp and message ordering
 
+If a user sends a new message before the messages are delivered that where sent while the user was offline, the new
```
Comment (adambabik) [inline status-payloads-spec.md]:
```suggestion
If a user sends a new message before the messages sent while the user was offline are received, the new
```

File: status-payloads-spec.md [hunk 2]
```diff
@@ -173,16 +173,27 @@ The following messages types MUST be supported:
...
+message is supposed to be displayed last in a chat. This is where the basic algorithm of Lamport timestamp would fall
```
Comment (adambabik) [inline status-payloads-spec.md]:
```suggestion
message is supposed to be displayed last in a chat. This is where the basic algorithm of Lamport timestamp would fall short
```

File: status-payloads-spec.md [hunk 3]
```diff
@@ -173,16 +173,27 @@ The following messages types MUST be supported:
...
+short, as it's only meant to order causally related events.
```
Comment (adambabik) [inline status-payloads-spec.md]:
```suggestion
as it's only meant to order causally related events.
```

File: status-payloads-spec.md [hunk 4]
```diff
@@ -173,16 +173,27 @@ The following messages types MUST be supported:
...
 
 `clock` SHOULD be calculated using the algorithm of [Lamport timestamps](https://en.wikipedia.org/wiki/Lamport_timestamps). When there are messages available in a chat, `clock`'s value is calculated based on the last received message in a particular chat: `max(timeNowInMs, last-message-clock-value + 1)`. If there are no messages, `clock` is initialized with `timestamp`'s value.
 
 Messages with a `clock` greater than `120` seconds over the whisper timestamp SHOULD be discarded, in order to avoid malicious users to increase the `clock` of a chat arbitrarily.
 
-Messages with a `clock` less than `120` might indicate an attempt to insert messages in the chat history, this is not distinguishable though from a re-transmit from the `datasync` layer. A client MAY mark this messages with a warning to the user, or discard them.
+Messages with a `clock` less than `120` seconds under the whisper timestamp might indicate an attempt to insert messages in the chat history, this is not distinguishable though from a re-transmit from the `datasync` layer. A client MAY mark this messages with a warning to the user, or discard them.
```
Comment (adambabik) [inline status-payloads-spec.md]:
```suggestion
Messages with a `clock` less than `120` seconds under the whisper timestamp might indicate an attempt to insert messages in the chat history which is not distinguishable from a `datasync` layer re-transit event. A client MAY mark this messages with a warning to the user, or discard them.
```

File: status-payloads-spec.md [hunk 5]
```diff
@@ -173,16 +173,27 @@ The following messages types MUST be supported:
 
 #### Clock vs Timestamp and message ordering
 
+If a user sends a new message before the messages are delivered that where sent while the user was offline, the new
+message is supposed to be displayed last in a chat. This is where the basic algorithm of Lamport timestamp would fall
+short, as it's only meant to order causally related events.
+
+The status client therefore makes a "bid", speculating that it will beat the current chat-timestamp, s.t. the status client's
+Lamport timestamp format is: `clock = `max({timestamp}{post-id}, chat_clock + 1)`
```
Comment (cammellos) [inline status-payloads-spec.md]: We only use `timestamp` now (in millisecond), `{post-id}` has been removed as not very useful.

File: status-payloads-spec.md [hunk 6]
```diff
@@ -173,16 +173,27 @@ The following messages types MUST be supported:
...
 
-Messages with a `clock` less than `120` might indicate an attempt to insert messages in the chat history, this is not distinguishable though from a re-transmit from the `datasync` layer. A client MAY mark this messages with a warning to the user, or discard them.
+Messages with a `clock` less than `120` seconds under the whisper timestamp might indicate an attempt to insert messages in the chat history, this is not distinguishable though from a re-transmit from the `datasync` layer. A client MAY mark this messages with a warning to the user, or discard them.
 
 `clock` value is used for the message ordering. Due to the used algorithm and distributed nature of the system, we achieve casual ordering which might produce counterintuitive results in some edge cases. For example, when one joins a public chat and sends a message before receiving the exist messages, their message `clock` value might be lower and the message will end up in the past when the historical messages are fetched.
 
+`post-id` is the id of a message.
```
Comment (cammellos) [inline status-payloads-spec.md]: Post id is not used anymore

**Review by cammellos — APPROVED** [review]
Looks good, the only comment is about post-id, which is not used anymore nor appended.

**Comment by oskarth** [comment]
@0kok0 Thanks for your PR! Could you have a look at above comments and update PR accordingly? Then we can merge this.

**Comment by 0kok0** [comment]
onit!

**Comment by 0kok0** [comment]
good for merge

**Comment by oskarth** [comment]
Thanks @0kok0!

**Pull request merged**